sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/DeserializerBuildHelper.scala

```diff
@@ -458,7 +460,10 @@ object DeserializerBuildHelper {
         Literal.create(provider(), ObjectType(classOf[Codec[_, _]])),
         "decode",
         ObjectType(tag.runtimeClass),
-        createDeserializer(encoder, path, walkedTypePath) :: Nil)
+        createDeserializer(encoder, path, walkedTypePath) :: Nil,
+        propagateNull = enc.nullable,
```

Comment:
Besides the fact that this produces more efficient code. Is there anyway this influences behavior? I am trying figure out if this needs testing.

Reply:
it's possible I just added for symmetry, it doesn't have an effect on the Spark test cases, I'll report back in a couple of hours if it affects any frameless ones - either adding a test to cover those two lines or reverting them.